wn/morphy.py

```diff
@@ -0,0 +1,306 @@
+
+"""
+Wordnet Morphy implementation
+"""
+
+import re
+from typing import Dict, Generator, Iterable, Iterator, Optional
+
+from wn._db import find_entries
+from wn.constants import NOUN, VERB, ADJ
+
+DETACHMENT_RULES = {
+    NOUN: [ # Can be converted to a dict if Python 3.7+ is guaranteed
+        ("s", ""),
+        ("ses", "s"),
+        ("xes", "x"),
+        ("zes", "z"),
+        ("ches", "ch"),
+        ("shes", "sh"),
+        ("men", "man"),
+        ("ies", "y"),
+    ],
+    VERB: [
+        ("s", ""),
+        ("ies", "y"),
+        ("es", "e"),
+        ("es", ""),
+        ("ed", "e"),
+        ("ing", "e"),
+        ("ing", ""),
+    ],
+    ADJ: [
+        ("er", ""),
+        ("est", ""),
+        ("er", "e"),
+        ("est", "e"),
+    ]
+}
+
+PREPOSITIONS = ["about", "above", "across", "after", "among", "around", "athwart", 
+                "at", "before", "behind", "below", "beneath", "beside", "besides", 
+                "between", "betwixt", "beyond", "but", "by", "during", "except", 
+                "for", "from", "into", "in", "near", "off", "of", "onto", "on", 
+                "out", "over", "since", "till", "to", "under", "until", "unto", 
+                "upon", "with"]
+
+def expand_regex_rules(rules: dict) -> dict:
```

Comment:
Functions that aren't meant to be called directly by users or outside of this module should be prefixed with `_`